### PR TITLE
Added a gallery example for saving & loading GenericMap using asdf

### LINKED ISSUE #431: Support for saving intermediate data products
Is there a good way to save intermediate SunPy data products? The pickle module does not work properly on some SunPy objects at the moment (e.g. maps, where the header information gets lost).

As an example, I may want to process some data only once (maybe it's time consuming) and generate some products which I then want to work with or manipulate many times, over multiple Python sessions.

### LINKED ISSUE #3961: Write example showing how to save GenericMap with asdf
As far as I can see #2366 added support to save a GenericMap using asdf. It would be good to add an example to this section: https://docs.sunpy.org/en/stable/generated/gallery/index.html#saving-and-loading-data showing how to do that.

Fixing this would close #431

### PR DESCRIPTION
**Pull request by chouhanaryan**
### Description

Added a gallery example that covers saving and loading GenericMaps using asdf.

Fixes #3961 
Fixes #431

### CODE REVIEW

File: examples/saving_and_loading_data/genericmap_in_asdf.py
Comment (Cadair) [inline examples/saving_and_loading_data/genericmap_in_asdf.py:10]:
```suggestion
community. It has deep integration with Python, SunPy and Astropy as well as
```

File: examples/saving_and_loading_data/genericmap_in_asdf.py
Comment (Cadair) [inline examples/saving_and_loading_data/genericmap_in_asdf.py:39]:
```suggestion
# need to put it into a dictionary. This becomes what asdf calls a "tree".
```

File: examples/saving_and_loading_data/genericmap_in_asdf.py
Comment (Cadair) [inline examples/saving_and_loading_data/genericmap_in_asdf.py:49]:
```suggestion
# Astropy, SunPy and asdf installed. We can reload it like so:
```

File: examples/saving_and_loading_data/genericmap_in_asdf.py
Comment (Cadair) [inline examples/saving_and_loading_data/genericmap_in_asdf.py:4]:
```suggestion
=======================================
Saving and loading sunpy Maps with asdf
=======================================
```

**Review by Cadair — APPROVED** [review]
Thanks for writing this, it looks great! A few minor comments :smile:

**Comment by Cadair** [comment]
This just needs a second review now :smile:

File: changelog/4494.docfix.rst
Comment (nabobalis) [inline changelog/4494.docfix.rst:1]:
```suggestion
Added a gallery example for saving and loading sunpy Maps using asdf.
```

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks again @chouhanaryan

**Pull request merged**
